sockeye/inference.py

```diff
@@ -952,20 +982,19 @@ def _get_inference_input(self, trans_inputs: List[TranslatorInput]) -> Tuple[mx.
         """
         bucket_key = data_io.get_bucket(max(len(inp.tokens) for inp in trans_inputs), self.buckets_source)
 
-        utils.check_condition(C.PAD_ID == 0, "pad id should be 0")
         source = mx.nd.zeros((len(trans_inputs), bucket_key, self.num_source_factors), ctx=self.context)
 
         for j, trans_input in enumerate(trans_inputs):
             num_tokens = len(trans_input)
-            source[j, :num_tokens, 0] = data_io.tokens2ids(trans_input.tokens, self.vocab_source)
-
-            utils.check_condition(self.num_source_factors == trans_input.num_factors,
-                                  "Unexpected number of factors for input %s. Expected %d" % (trans_input,
-                                                                                              self.num_source_factors))
-            for i, factor in enumerate(trans_input.factors):
-                utils.check_condition(len(factor) == num_tokens,
-                                      "Factor sequence has different length than tokens: %s" % trans_input)
-                source[j, :num_tokens, i + 1] = data_io.tokens2ids(factor, self.source_factor_vocabs[i])
+            source[j, :num_tokens, 0] = data_io.tokens2ids(trans_input.tokens, self.source_vocabs[0])
+
+            factors = trans_input.factors if trans_input.factors is not None else []
+            num_factors = 1 + len(factors)
+            if num_factors != self.num_source_factors:
+                logger.warning("Input has not enough factors, %d, but expected %d",num_factors, self.num_source_factors)
+            for i, factor in enumerate(factors[:self.num_source_factors - 1], start=1):
```

Comment:
Wouldn't this fail if num_factors < self.num_source_factors?

Reply:
This is fine in python slicing:
```
In [2]: list(range(2))[:14]
Out[2]: [0, 1]
```

Reply:
Right, that is slicing and not accessing!